core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -983,8 +979,14 @@ protected void onException(Channel channel, Exception e) throws IOException {
      */
     protected abstract void closeChannels(List<Channel> channel) throws IOException;
 
-
-    protected abstract void sendMessage(Channel channel, BytesReference reference, Runnable sendListener) throws IOException;
+    /**
+     * Sends message to channel. The listener will be called with an exception if the send operation throws an exception. If the operation
+     * is successful, the listener will be called with null.
```

Comment:
I guess, but I wonder why not have the listener define `onSuccess` and `onFailure`?